### PR TITLE
replace options resolver into symfony config component

### LINKED ISSUE #19: input sanitation of `files` config variables
_From @tolry on November 29, 2014 23:58_

e.g. phpcs/phpmd just get those parameters forwarded, they are not checked for being relative paths

_Copied from original issue: simplethings/SimpSpector#39_

### PR DESCRIPTION
**Pull request by DavidBadura**
the config component is cleaner and more powerfull.
you can better validate, normalize and dump gadget configuration reference.

fixes #19

### CODE REVIEW

**Comment by DavidBadura** [comment]
ready

File: src/Gadget/AbstractGadget.php [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\OptionsResolver\Options;
-use Symfony\Component\OptionsResolver\OptionsResolver;
-
-/**
- * @author David Badura <d.a.badura@gmail.com>
- * @author Tobias Olry <tobias.olry@gmail.com>
- * @author Lars Wallenborn <lars@wallenborn.net>
- */
-abstract class AbstractGadget implements GadgetInterface
```
Comment (tolry) [inline src/Gadget/AbstractGadget.php]: finally :wink:

**Comment by DavidBadura** [comment]
todo: fix tests

**Comment by larsborn** [comment]
Ah you just showed this changes to me in person. I totally support a merge!

**Comment by DavidBadura** [comment]
ready :wink:

**Comment by tolry** [comment]
I merge, since I also give a :+1: for this :wink:

**Pull request merged**